caput/mpiarray.py

```diff
@@ -843,38 +864,37 @@ def allreduce(self, op=None):
 
         Parameters
         ----------
-        op : MPI reduction operation
-            Reduction operation to perform. Default: MPI.SUM
+        op : MPI.Op | None
+            Reduction operation to perform. Default: :py:func:`~mpi4py.MPI.SUM`.
 
         Returns
         -------
-        scalar
-            result of reduction
+        value : number
+            Result of reduction. This is a single value which is the
+            same on all ranks.
         """
-        from mpi4py import MPI
-
         if self.local_shape != (1,):
             raise ValueError(
                 "MPIArray.allreduce()'s is limited to arrays with a single element per "
                 f"rank, but rank={self.comm.rank} has shape-{self.local_shape}."
             )
 
         result_arr = np.zeros((1,), dtype=self.dtype)
-        self.comm.Allreduce(self, result_arr, MPI.SUM if op is None else op)
+        self.comm.Allreduce(self, result_arr, mpitools.SUM if op is None else op)
 
         return result_arr[0]
 
-    def enumerate(self, axis):
+    def enumerate(self, axis: int) -> Iterator[tuple[int, int]]:
         """Helper for enumerating over a given axis.
 
         Parameters
         ----------
-        axis : integer
-            Which access to enumerate over.
+        axis : int
+            Axis to enumerate over.
 
         Returns
         -------
-        iterator : (local_index, global_index)
+        enumerator : Iterator
```

Comment:
```suggestion
        enumerator : enumerate
```
`enumerate()` is a class.  This function is returning an instance of it.